### PR TITLE
feat: add children action on Android

### PR DESCRIPTION
**Pull request by viniciusguardieirozup**
### Related Issues

#272 

### Description and Example

Create a addChildrenAction to add a list of ServerDrivenComponent on the views tree
### Checklist

Please, check if these important points are met using `[x]`:

- [x] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [x] I avoided _breaking changes_ by not changing public APIs that people rely on.
- [x] I am willing to follow-up on review comments in a timely manner.

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: backend/widgets/src/main/kotlin/br/com/zup/beagle/widget/action/AddChildrenAction.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package br.com.zup.beagle.widget.action
+
+import br.com.zup.beagle.core.ServerDrivenComponent
+
+enum class Mode {
+    APPEND, PREPEND, REPLACE
```
Comment (botbeaglezup) [inline backend/widgets/src/main/kotlin/br/com/zup/beagle/widget/action/AddChildrenAction.kt:22]: <table data-meta="generated_by_backend">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/36306e8f5d6a310d5ca4f81a88ec76232cacace2/backend/widgets/src/main/kotlin/br/com/zup/beagle/widget/action/AddChildrenAction.kt#L22"></span>Enum entry names should match the pattern: [A-Z][_a-zA-Z0-9]*</td>
    </tr>
  </tbody>
</table>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/36306e8f5d6a310d5ca4f81a88ec76232cacace2/backend/sample/core/src/main/kotlin/br/com/zup/beagle/sample/builder/ButtonScreenBuilder.kt#L29">backend/sample/core/src/main/kotlin/br/com/zup/beagle/sample/builder/ButtonScreenBuilder.kt#L29</a> - br.com.zup.beagle.widget.action.* is a wildcard import. Replace it with fully qualified imports.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_backend">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: android/beagle/src/main/java/br/com/zup/beagle/android/action/AddChildrenAction.kt [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    var value: List<ServerDrivenComponent>,
+    var mode: Mode? = Mode.APPEND
+) : Action {
+    override fun execute(rootView: RootView, origin: View) {
+        try {
+            val view = (rootView.getContext() as AppCompatActivity).findViewById<ViewGroup>(componentId.toAndroidId())
+            val viewList = convertServerDrivenListOnViewList(value, rootView)
+            addValueToView(view, viewList)
+        } catch (exception: Exception) {
+            BeagleLoggerDefault().error("This view cannot receive children")
```
Comment (tuliopereirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/action/AddChildrenAction.kt:43]: Use BeagleMessageLogger instead.

**Pull request merged**